src/main.ts

```diff
@@ -49,10 +52,34 @@ async function run() {
     const number = context.payload.pull_request.number;
     const event = status > 0 ? "REQUEST_CHANGES" : "COMMENT";
     const body = [
-      `## [size-limit](${SIZE_LIMIT_URL}) report`,
+      SIZE_LIMIT_HEADING,
       table(limit.formatResults(base, current))
     ].join("\r\n");
 
+    if (ignoreIdentical) {
+      try {
+        const { data: reviews } = await octokit.pulls.listReviews({
+          ...context.repo,
+          // eslint-disable-next-line camelcase
+          pull_number: number
+        });
+
+        const sizeLimitReviews = reviews.filter(review =>
+          review.body.startsWith(SIZE_LIMIT_HEADING)
+        );
+        // only look at the last review (it could return to a previous state which we would want to know about)
+        if (
+          sizeLimitReviews.length > 0 &&
+          sizeLimitReviews[sizeLimitReviews.length - 1].body === body
+        ) {
+          // The last review was the same, so we shouldn't repeat
+          return;
+        }
```

Comment:
If we go for the update approach. We will need to use `find` instead of filter and then do something like this:
```javascript
octokit.pulls.updateReview({
  owner,
  repo,
  pull_number,
  review_id, // here we can use the id if there is one
  body,
});
```
Let me know your thoughts. Thank you again!

Reply:
I think in this scenario, due to some of the points I mentioned in my comment
> Can't update the event so a reject will always be a reject, and will still need a second review in the case of that switch

You'd still need to get the last of potentially many. I'm also happy to explore with you how to test this since it's not the easiest behavior to verify.